Homework 18

TetianaZaikina opened the pull request:
Hello, my teacher!
Please, have enough patience to look at it!
Thanks for your time :)

TetianaZaikina committed:
Update LoginTests.java

TetianaZaikina committed:
Merge branch 'main' of https://github.com/TetianaZaikina/A49-QA-Automation-Framework

TetianaZaikina committed:
Homework18

duykhanhrc commented on src/test/java/Homework18.java:
```diff
@@ -0,0 +1,42 @@
+import org.testng.annotations.Test;
+
+public class Homework18 extends BaseTest{
+    @Test()
```
Hi,
I have checked your homework 18, but there are a lot of errors in your code that make the test script unable to execute. Please check it carefully again and correct it.
Best wishes from the HW team (Khanh, khanh@testpro.io)

Here are some of my suggestions:
1. You need to use correct annotations @Test testng, @Test not is @Test()

duykhanhrc commented on src/test/java/Homework18.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    private void checkSongIsPlaying() {
+        WebElement soundBar= driver.findElement(By.cssSelector("[data-testid='sound-bar-play']"));
```
2. I don't see you define step initialize driver in this Homework18 script or `BaseTest `class, so can't perform any action on any element if driver is not initialized and lunch browser.
//HINT: You can define step launchBrowser() into BaseTest after all classes inheritance from `BaseTest `driver will be initialized

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/44592718/2564f3f7-cbe4-4398-bff5-724c2d24ed23)

duykhanhrc commented on src/test/java/Homework18.java:
```diff
@@ -0,0 +1,42 @@
+
+    private void provideEmail(String "pchelka240386@gmail.com"String mail) {
+    }
+
```
syntax error and not define element for email filed
//HINT: 

```
private void provideEmail(String mail) {
        WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
        emailField.click();
        emailField.sendKeys(mail);
    }
```

duykhanhrc commented on src/test/java/Homework18.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    private void providePassword(String te$t$tudent) {
+    }
```
please check the same comment on the previous provideEmail function

duykhanhrc commented on build.gradle:
```diff
@@ -15,8 +15,13 @@ dependencies {
-    }
-}
\ No newline at end of file
+
```
missing quotes at here
//HINT:
```
test {
    useTestNG() {
        suites "src/test/resources/TestNG.xml"
    }
}
```

duykhanhrc commented on src/test/resources/Homework18.xml:
```diff
@@ -0,0 +1,16 @@
+        <classes>
+            <class name="Homework18"/>
+                <methods>
+                      <include name="playSong"/>
```
You need to include correct function name of class name to test. And in the your `Homewrok18 `class above, I can see you are defining function name test is  `PlaySong` is not `playSong`
Please upate it

duykhanhrc commented on src/test/java/LoginTests.java:
```diff
@@ -22,5 +22,6 @@ public void loginEmptyEmailPassword() {
         Assert.assertEquals(driver.getCurrentUrl(), url);
         driver.quit();
+        +
     }
```
remove the + character here

duykhanhrc commented on src/test/java/LoginTests.java:
```diff
@@ -22,5 +22,6 @@ public void loginEmptyEmailPassword() {
         driver.get(url);
         Assert.assertEquals(driver.getCurrentUrl(), url);
         driver.quit();
+        +
```
remove the + character here

TetianaZaikina committed:
Homework18

-corrected the mistakes and applied changes
- tried to stick to Natalia's file on GitHub
-the code is not working

TetianaZaikina commented:
Good morning, Pham Duy Khanh!
Thank you for checking my homework and pointing out to mistakes. I have applied the changes but the code still did not run. I started to compare my work with Natalia's file on GitHub and (i guess) got more mistakes than I had. The works differ in structure and it misleads me. I am stuck on lesson 17 and 18 so far.
Best,
Tetiana

TetianaZaikina committed:
Homework18

It passed!
Thank you!!!

duykhanhrc commented:
Hi friend,
Your homework 18 has been reviewed, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Khanh,  khanh@testpro.io)!

Test suite Homework18 Passed
![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/44592718/9f4d9345-816a-4af3-9305-6a496e2786f3)
